keanu-python/keanu/algorithm/sampling.py

```diff
@@ -110,7 +110,8 @@ def sample(net: BayesNet,
         probabilistic_model.unwrap(), vertices_unwrapped, draws).drop(drop).downSample(down_sample_interval)
 
     vertex_samples = {
-        Vertex._get_python_label(vertex_unwrapped): list(
+        # label can't be None. See __check_if_vertices_are_labelled
+        cast(str, Vertex._get_python_label(vertex_unwrapped)): list(
```

Comment:
`cast` is the a very blunt tool for turning an `Optional[str]` into a `str`. It's also not clear to the reader that `cast` is just a `typing` thing that does not affect the functionality.

How about instead `__check_if_vertices_are_labelled` returns a map of labels (i.e. the values are `str` not `Optional[str]`? Then you don't need to call `_get_python_label` here.

Reply:
Good idea. I like that